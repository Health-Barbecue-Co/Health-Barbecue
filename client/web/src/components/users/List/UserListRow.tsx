import React from 'react'
import { TableRow, TableCell, Button } from '@material-ui/core'
import { Link } from 'react-router-dom'
import { IUser } from '../../../models/user'
import { IColAction } from './IColAction'

type UserListRowProps = {
  user: IUser
  actionCol: IColAction[]
}

export const UserListRow: React.FC<UserListRowProps> = (
  props: UserListRowProps
) => {
  const { user, actionCol } = props

  if (!user) {
    return null
  }

  return (
    <TableRow>
      <TableCell component="th" scope="row">
        {user.firstname}
      </TableCell>
      <TableCell align="right">{user.lastname}</TableCell>
      <TableCell align="right">{user.login}</TableCell>
      <TableCell align="right">{user.role}</TableCell>
      <TableCell align="right">
        {actionCol.map((action) => (
          <Button component={Link} to={action.link} key={action.link}>
            {action.icon}
          </Button>
        ))}
      </TableCell>
    </TableRow>
  )
}
