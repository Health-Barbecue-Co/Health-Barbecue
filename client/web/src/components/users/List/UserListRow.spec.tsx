import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import { UserListRow } from './UserListRow'

describe('UserListRow', () => {
  it('user attribute is mandatory', () => {
    try {
      mount(<UserListRow />)
    } catch (e) {
      expect(e).toBeDefined()
    }
  })

  it('a table row with user information', () => {
    const user = {
      id: 'myId',
      firstname: 'firstname',
      lastname: 'lastname',
      login: 'login',
      role: 'role',
    }
    const wrapper = renderer
      .create(<UserListRow user={user} actionCol={[]} />)
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
