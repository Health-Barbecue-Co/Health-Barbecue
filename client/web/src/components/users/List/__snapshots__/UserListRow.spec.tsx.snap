// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListRow a table row with user information 1`] = `
<tr
  className="MuiTableRow-root"
  role={null}
>
  <th
    aria-sort={null}
    className="MuiTableCell-root"
    role="cell"
    scope="row"
  >
    firstname
  </th>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    lastname
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    login
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight"
  >
    role
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  />
</tr>
`;
